addons/website_slides/static/src/js/slides_course_join.js

```diff
@@ -39,25 +73,35 @@ var CourseJoinWidget = publicWidget.Widget.extend({
             }
         }).popover('show');
     },
+    /**
+     * @private
+     */
+    _signInAndJoinCourse: function () {
+        var url = this._createLoginRedirectUrl();
+        this.trigger_up('before_join_course_redirect', {url: url});
```

Comment:
Sorry but I changed my mind about this.

I think the parent widget should send a new option to the CourseJoinWidget: `beforeJoin` that is a Promise.
And this code could do:
`this.beforeJoin.then(function() { // compute url and redirect });`

And the parent would init with:
- If they don't need to do anything before the redirect: 
`beforeJoin: Promise.resolve()`
- If they need to do an _rpc before the redirect:
```
beforeJoin: this._saveQuizAnswers.bind(this)

...

_saveQuizAnswers: function () {
    return self._rpc(...):
}
```

I think it will be much cleaner that way.

Again, sorry I changed my mind about this thing, but it's not trivial so we should experiment a bit and see what the best solution is.